src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -58,33 +58,72 @@ type writeCommitLogFn func(
 ) error
 type commitLogFailFn func(err error)
 
-type completionFn func(err error)
+type valueTypeFn func(f File, err error)
```

Comment:
For better reuse, which we'll need in the future, can we just rename this to `callbackFn`?

And maybe the first arg it would be better to be some composite event struct, so we can add cleanly to it in the future:

```go
func callbackFn func(result callbackResult)

type callbackResultType uint

const (
  activeLogsCallback callbackResultType = iota
  // ... more in the future
)

type callbackResult struct {
  resultType callbackResultType
  err error
  activeLogs activeLogsCallbackResult // to be used if and only if eventType == activeLogsResultCallbackEventType
}

type activeLogsCallbackResult struct {
  file *File
}

func (r callbackResult) activeLogsCallbackResult() (activeLogsCallbackResult, error) {
  if expectedType := r.resultType; expectedType != activeLogsCallback {
    return activeLogsCallbackResult{}, fmt.Errorf("wrong result type: expected=%d, actual=%d", expectedType, r.resultType)
  }
  if r.err != nil {
    return activeLogsCallbackResult{}, err
  }
  return r.activeLogs, nil
}

// Now from calling code:
func foo() {
  var (
    result activeLogsCallbackResult
    err error
  )
  writes <- commitLogWrite{
    valueType: activeLogsValueType,
    callbackFn: func(r callbackResult) {
      result, err = r.activeLogsCallbackResult()
      wg.Done()
    },
  }
  if err != nil {
    return nil, err
  }
  return result.file, nil
}
```

Reply:
Yeah I can make this change, the reason I didn't is that the Rotate() API will need pretty much the same result as the ActiveLogs one but this is fine too

Reply:
`resultType` seems a little overkill, I think I can just repurpose the `valueType` for now to infer the result type

Reply:
`resultType` seems a little overkill, I think I can just repurpose the `valueType` for now to infer the result type.

Reply:
done